### DISCUSSION TITLE
Restrict picklist values to profiles

### DISCUSSION THREAD

**Discussion by yeti92force** [YetiForceCompany/YetiForceCRM#16739] [Feature improvement]
Hi team,

It would be very useful to restrict picklist values ​​to specific profiles.

Regards,
Mathieu Skz

**Comment by vovpff** [comment]
Let's use role based values? But in this case you need more roles in your system

**Reply by yeti92force** [reply]
It's possible to hide picklist values with rôles in yetiforce ??! Any tutorial please ? Its à high priority for our business

**Reply by vovpff** [reply]
There is an option when you create pick list. Picklist can be usual (available for all users) or role based. In case role based you can tune roles what can access to pick list values. This is difficult to describe. You need to try it by youself

**Reply by yeti92force** [reply]
Nice!  
We create our picklist values with soql queries. Is there any way to add picklist values restrictions from rôle ?  Or maybe do you kwno the queries the query to create a picklist value with its association to a role? 

Regards

**Comment by rskrzypczak** [comment]
Picklist configuration panel:
![image](https://user-images.githubusercontent.com/10221999/208047472-a7c67536-fd20-4a95-8a87-fa39db10f7a1.png)

Helpful video:
 [![Watch the video](https://img.youtube.com/vi/J14paUYtVwI/default.jpg)](https://www.youtube.com/watch?v=J14paUYtVwI)

**Reply by yeti92force** [reply]
Thanks. Is it possible to create multiple picklist values with based rôle restriction whith soql queries ? Our business model use a high volume of values, so we cant do manual click to setup the role one by one..!

**Reply by rskrzypczak** [reply]
Adding picklist values via sql queries is the worst possible option.
If you want to add picklist values in script, do it through objects.
e.g.: https://github.com/YetiForceCompany/YetiForceCRM/blob/d91dc1c6b46aa0991110fa88f116dd3b155c5806/vtlib/Vtiger/Field.php#L33

**Reply by rskrzypczak** [reply]
Besides, you can import piclkist values from a file.
![image](https://user-images.githubusercontent.com/10221999/208051682-de23714d-b6ca-4558-a206-82f61deb6e7c.png)
